src/models/gamma_poisson_model.py

```diff
@@ -177,19 +177,23 @@ def _knreach(self, k, n, I, Imax, alpha, beta):
         where C(n, k) is the binomial coefficient n!/[k!(n-k)!].
 
         Args:
-          k:  Scalar number of impressions that were seen by the user.
-          n:  Available inventory for the user.
-            This can be either a scalar or a numpy array.
+          k:  (C, ) ndarray. Numbers of impressions that were seen by the user.
+          n:  (M, ) ndarray. Available inventory for the user.
           I:  Total number of impressions shown to all users.
           Imax: Total size of impression inventory.
           alpha: Parameter of Gamma-Poisson distribution.
           beta: Parameter of Gamma-Poisson distribution.
         Returns:
-          Probability that a randomly chosen user will have an inventory
-          of n impressions, of which k are shown.
+          (C, M) ndarray. Probability that a randomly chosen user will have an
+          inventory of n impressions, of which k are shown.
         """
-        kprob = scipy.stats.binom.logpmf(k, n, I / Imax)
-        return np.exp(kprob + self._logpmf(n, alpha, beta))
+        binom_dist = scipy.stats._discrete_distns.binom_gen(name="binom_dist")
```

Comment:
Did you test the idea of creating a binom_dist object in the constructor?

Reply:
Yes. I just got the result, and it looks good. I'll share the result with you and change the code to use `self.binom_dist`.